cmd/gaia/app/genesis.go

```diff
@@ -152,7 +152,7 @@ func GaiaAppGenTxNF(cdc *wire.Codec, pk crypto.PubKey, addr sdk.AccAddress, name
 
 	validator = tmtypes.GenesisValidator{
 		PubKey: pk,
-		Power:  freeFermionVal,
+		Power:  freeFermionVal.Int64(),
```

Comment:
I believe the choice was so that we're consistent with Tendermint where validator's power is `int64`. There is a risk of overflow when you add them together (AddTokensFromDel), but as long as total amount is less than 9223372036854775807 we're fine. I'll leave the final decision to cosmos sdk core devs however.

Reply:
If I understand correctly, melekes is suggesting leaving `freeFermionVal` as an int64. I agree with this

Reply:
Got your point. I have changed the code according to your suggestion. Previously, I think it seems weird to use different data type to represent token related variables. However, overflow may cause crash which should be avoid. Thanks for your thoughtful suggestion.